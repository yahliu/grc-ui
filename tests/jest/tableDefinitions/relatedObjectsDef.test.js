/* Copyright (c) 2021 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'
import {
  buildViewYamlLink
} from '../../../src-web/tableDefinitions/relatedObjectsDef'

describe('tableDefinitions relatedObjectsDef - buildViewYamlLink', () => {
  it('should return valid yaml link with namespace', () => {
    const item = {
      cluster: 'testCluster',
      object: {
        apiVersion: 'testApiVersion',
        kind: 'testKind',
        metadata: {
          name: 'testName',
          namespace: 'testNamespace'
        }
      }
    }
    expect(buildViewYamlLink(item, 'us-en')).toMatchSnapshot()
  })

  it('should return valid yaml link without namespace', () => {
    const item = {
      cluster: 'testCluster',
      object: {
        apiVersion: 'testApiVersion',
        kind: 'testKind',
        metadata: {
          name: 'testName',
          namespace: ''
        }
      }
    }
    expect(buildViewYamlLink(item, 'us-en')).toMatchSnapshot()
  })

  it('should return empty yaml link', () => {
    const item = {
      cluster: '',
      object: {
        apiVersion: '',
        kind: '',
        metadata: {
          name: '',
          namespace: ''
        }
      }
    }
    expect(buildViewYamlLink(item, 'us-en')).toMatchSnapshot()
  })
})
