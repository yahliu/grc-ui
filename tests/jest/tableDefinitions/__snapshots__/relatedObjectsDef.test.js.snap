// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tableDefinitions relatedObjectsDef - buildViewYamlLink should return empty yaml link 1`] = `""`;

exports[`tableDefinitions relatedObjectsDef - buildViewYamlLink should return valid yaml link with namespace 1`] = `
<a
  href="/resources?cluster=testCluster&kind=testKind&apiversion=testApiVersion&namespace=testNamespace&name=testName"
  rel="noopener noreferrer"
  target="_blank"
>
  View yaml
</a>
`;

exports[`tableDefinitions relatedObjectsDef - buildViewYamlLink should return valid yaml link without namespace 1`] = `
<a
  href="/resources?cluster=testCluster&kind=testKind&apiversion=testApiVersion&name=testName"
  rel="noopener noreferrer"
  target="_blank"
>
  View yaml
</a>
`;
